tests/test_util.py

```diff
@@ -78,3 +83,42 @@ def test_copy_file_bulk(self) -> None:
                             call("/src/path/foo/file2", "/target/path/foo/file2"),
                         ]
                     )
+
+
+def _response_generator(target_value: int) -> Callable[[Request], Response]:
+    """Return a generator with closure so a value can be tracked across multiple invocations."""
+    value = 0
+
+    def generator(request: Request) -> Response:  # pylint: disable=unused-argument
+        """Add the next value as a header and adjust the status code based on the value."""
+        nonlocal value, target_value
+        value += 1
+        response = Response()
+        response.status_code = 403 if value <= (target_value + 1) else 200
+        response.headers["X-VALUE"] = str(value)
+        return response
+
+    return generator
+
+
+def test_get_http_failure(httpserver: HTTPServer) -> None:
+    """Test get http operations when a 403 error code is received."""
+    # Set up a localhost URL.
+    mocked_url = httpserver.url_for("")
+
+    # Retrieve the allowed number of retries on a failed request.
+    target_value = defaults.getint("requests", "error_retries", fallback=5)
+
+    # Create and assign the stateful handler.
+    handler = _response_generator(target_value)
+    httpserver.expect_request("").respond_with_handler(handler)
+
+    # Assert the request fails and returns nothing.
+    assert send_get_http_raw(mocked_url) is None
+
+    # Test for a correct response after the expected number of retries and other requests (including this one).
```

Comment:
I'm not sure I follow this part of the test. Doesn't it make more sense to test (in a separate function) that the request succeeds if the response does not have an error code within the allowed number of attempts?